metadata.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+package SMPS;
+
+option go_package = "spec;spec";
+
+message ApplicationMetadata {
+
+    enum application_protocol {
+        http = 0;
```

Comment:
Eventually, we'll want this to refer to the traffic.proto - https://github.com/layer5io/service-mesh-performance-specification/issues/38

And this, too - https://github.com/layer5io/service-mesh-performance-specification/blob/master/smps.proto#L13

Reply:
Makes sense, do you want me to do this in the same PR ?

Reply:
Thanks. No, we can circle back here.

Reply:
@mandarjog @carolynhu have you done any performance testing comparing efficiencies / handling of gRPC vs Nats requests?

Reply:
no, we did not compare those yet.

Reply:
Oh, I see. Thank you for confirming, @carolynhu.